src/System.Net.Http/tests/Performance/Perf.SocketsHttpHandler.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Net.Http.Tests
+{
+    public sealed class SocketsHttpHandlerPerfTest
+    {
+        const int InnerIterationCount = 1000;
+
+        public static IEnumerable<object[]> Get_MemberData() =>
+            from ssl in new[] { false, true }
+            from connectionPerRequest in new[] { false, true }
+            from chunkedResponse in new[] { false, true }
+            from responseLength in new[] { 1, 100_000 }
+            select new object[] { ssl, connectionPerRequest, chunkedResponse, responseLength };
+
+        [Benchmark(InnerIterationCount = InnerIterationCount)]
+        [MemberData(nameof(Get_MemberData))]
+        public async Task Get(bool ssl, bool connectionPerRequest, bool chunkedResponse, int responseLength)
+        {
+            using (var serverCert = System.Net.Test.Common.Configuration.Certificates.GetServerCertificate())
+            using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                #region Server
+                listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                listener.Listen(int.MaxValue);
+                ReadOnlyMemory<byte> responseBytes = Encoding.UTF8.GetBytes(chunkedResponse ?
+                    $"HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n{responseLength.ToString("X")}\r\n{new string('a', responseLength)}\r\n0\r\n\r\n" :
+                    $"HTTP/1.1 200 OK\r\nContent-Length: {responseLength}\r\n\r\n{new string('a', responseLength)}");
```

Comment:
You should add an 'if' check for 'connectionPerRequest' and send back a response header such as "Connection: close" if you're really then closing the TCP connection on the server side.  That response header is what clients use to determine what they should do with their current connection.  It's more of an interesting test than having the client send "Connection: close" (which is really just a request to the server advising it that it doesn't support/want a persistent connection".  In most actual real-world scenarios, the client is expecting a persistent connection; it won't even send "Connection: keep-alive" since that the is default for HTTP/1.1.  But since you're testing both "Connection: keep-alive" and "Connection: close" scenarios, it would be best to have this loopback server act like a real server and sending back "Connection: close" is required by the RFC if the server is going to close the connection.  It also will help test the SocketsHttpHandler to see how it is handling getting that header back in the response (i.e. it should close the TCP connection on its side).